src/shared/cache/SampleListsInStudyCache.ts

```diff
@@ -0,0 +1,39 @@
+import LazyMobXCache, {AugmentedData} from "../lib/LazyMobXCache";
+
+import client from "../api/cbioportalClientInstance";
+import * as _ from "lodash";
+import { SampleList } from "shared/api/generated/CBioPortalAPI";
+
+function queryToKey(studyId:string) {
+    return studyId;
+}
+
+function dataToKey(molecularProfiles:SampleList[], studyId:string) {
+    return studyId;
+}
+
+async function fetch(studyIds:string[]):Promise<AugmentedData<SampleList[], string>[]> {
+
+    let promises = studyIds.map(studyId => client.getAllSampleListsInStudyUsingGET({
+        studyId: studyId,
+        projection: 'DETAILED'
+    }))
+
+    const sampleLists:SampleList[] = await Promise.all(promises).then((allData) => {
+        return _.flatMap(allData)
+    })
+    const sampleListsByStudy = _.groupBy(sampleLists, profile=>profile.studyId);
+    return studyIds.map(studyId=>{
+        const data = [(sampleListsByStudy[studyId] || [])];
+        return {
+            data,
+            meta: studyId
```

Comment:
I would do { meta:  { studyId } } as  meta gives no info about what that property is